src/main/scala/lantern/TensorDifferentiation.scala

```diff
@@ -456,6 +459,14 @@ trait TensorDsl extends DslOps with Diff {
             val offres = var_new[Int](offRes)
             val offx = var_new[Int](offX)
             val offy = var_new[Int](offY)
+
+            // Handle scalar tensor case.
```

Comment:
The elementwise broadcasting helpers were broken by the inclusion of scalar tensors.
I tried to fix them without adding special case logic for scalar tensors, but couldn't figure out it.
Reinvestigate later.